src/Features/CSharp/Portable/GenerateMember/GenerateVariable/CSharpGenerateVariableService.cs

```diff
@@ -23,7 +25,7 @@ protected override bool IsExplicitInterfaceGeneration(SyntaxNode node)
             => node is PropertyDeclarationSyntax;
 
         protected override bool IsIdentifierNameGeneration(SyntaxNode node)
-            => node is IdentifierNameSyntax identifier && !identifier.Identifier.CouldBeKeyword();
+            => node is IdentifierNameSyntax identifier && ClassificationHelpers.GetClassification(node.GetFirstToken()) == ClassificationTypeNames.Identifier;
```

Comment:
I'm surprised this is a sufficient helper.  I know classification needs some semantics to figure out if some identifiers should be keywords or not.  For example, for figuring out if 'value' is an identifier or should be thought of as a keyword.

Reply:
If this isn't a sufficient helper than perhaps this should be moved out of "Small Fixes".  It would take a bit of logic to determine when a contextual keyword can be an identifier, though it would be a useful utility to have.  @sharwell , what do you think?

Reply:
@CyrusNajmabadi what exactly do you mean by semantics? Isn't it the case that contextual keywords are determined purely by syntax?

Reply:
Another though: if we used a contextual keyword (that happens to be a keyword in this context), would we get the same error message (CS0103 The name 'X' does not exist in the current context)? My guess is not, and if not, maybe there is no additional check needed after all.

Reply:
There are several cases where there are things though of a "keywords" which are neither "reserved words" nor "contextual reserved words".  `value`, `var`, `unamanged`, `dynamic`, `nameof`, etc. are examples of these.

They are *pure* identifiers (lexically and syntactically).  However, the language imbues them with meaning if they don't happen to bind to anything.  That's why you need semantics.  Lexical/syntactical analysis is insufficient to tell you.

Reply:
@CyrusNajmabadi Yes I know, I thought we still wouldn't allow generating variables for them, only for actual syntactical contextual keywords.

Reply:
I didn't know `value` fell into that category though...

Reply:
> @CyrusNajmabadi Yes I know, I thought we still wouldn't allow generating variables for them

What stops us from generating variables for them?